flowmachine/flowmachine/features/location/labelled_spatial_aggregate.py

```diff
@@ -0,0 +1,117 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from typing import List
+
+from flowmachine.core import Query
+from flowmachine.core.mixins import GeoDataMixin
+from flowmachine.features.location.spatial_aggregate import SpatialAggregate
+
+
+class LabelledSpatialAggregate(GeoDataMixin, Query):
+    """
+    Class representing a disaggregation of a SpatialAggregate by some label or set of labels
+
+    Parameters
+    ----------
+    locations: Query
+        Any query with a subscriber and location columns
+    subscriber_labels: Query
+        Any query with a subscriber column
+    label_columns: List[str]
+        A list of columns in subscriber_labels to aggregate on
+
+    Examples
+    --------
+
+    >>> locations = locate_subscribers(
+    ...     "2016-01-01",
+    ...     "2016-01-02",
+    ...     spatial_unit=make_spatial_unit("admin", level=3),
+    ...     method="most-common",
+    ... )
+    >>> metric = SubscriberHandsetCharacteristic(
+    ...     "2016-01-01", "2016-01-02", characteristic="hnd_type"
+    ... )
+    >>> labelled = LabelledSpatialAggregate(locations=locations, subscriber_labels=metric)
+    >>> labelled.get_dataframe()
+               pcod label_value  value
+     0  524 3 08 44     Feature     36
+     1  524 3 08 44       Smart     28
+     2  524 4 12 62     Feature     44
+     3  524 4 12 62       Smart     19'
+
+    """
+
+    def __init__(
+        self,
+        locations: Query,
+        subscriber_labels: Query,
```

Comment:
I'd imagine that in practice, if we wanted to use multiple labels they would usually come from multiple queries. But I suppose there's always the option of `join`ing the queries first, to get a single query to use as the `subscriber_labels`, so I think this covers that use-case too.